### ISSUE TITLE
Build successful, but traceback says //mediapipe_api\mediapipe_proto_srcs.zip is not a zip file

### ISSUE THREAD

**Issue by zigzagoon1** [homuler/MediaPipeUnityPlugin#614]
### Plugin Version or Commit ID

v0.10.0

### Unity Version

2021.3.3f1

### Your Host OS

Windows 11

### Target Platform

UnityEditor

### [Windows Only] Visual Studio C++ and Windows SDK Version

Windows SDK: 10.0.19041.0
 

### Command Sequences

ran build command (*path to python.exe* build.py build --desktop cpu --opencv=cmake -vv

### Log

C:\Users\kmark\UnityProjects\MediaPipeUnityPlugin-master>C:\Users\kmark\AppData\Local\Programs\Python\Python310\python.exe build.py build --desktop cpu --opencv=cmake -vv
INFO (build.py): Building protobuf sources...
DEBUG (build.py): Running `bazel --output_user_root C:/_bzl build -c opt --action_env PYTHON_BIN_PATH="C://Users//kmark//AppData//Local//Programs//Python//Python310//python.exe" --action_env ProgramData --action_env PROCESSOR_ARCHITECTURE --action_env PROCESSOR_IDENTIFIER --action_env PROCESSOR_LEVEL --action_env PROCESSOR_REVISION --verbose_failures //mediapipe_api:mediapipe_proto_srcs`
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'com_google_absl' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'com_google_benchmark' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'pybind11_bazel' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'com_google_protobuf' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'com_google_googletest' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'com_github_gflags_gflags' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'rules_python' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'build_bazel_rules_apple' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'build_bazel_rules_swift' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'build_bazel_apple_support' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'xctestrunner' because it already exists.
DEBUG: C:/_bzl/k57vclnn/external/org_tensorflow/third_party/repo.bzl:124:14:
Warning: skipping import of repository 'pybind11' because it already exists.
INFO: Analyzed target //mediapipe_api:mediapipe_proto_srcs (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //mediapipe_api:mediapipe_proto_srcs up-to-date:
  bazel-bin/mediapipe_api/mediapipe_proto_srcs.zip
INFO: Elapsed time: 3.705s, Critical Path: 0.02s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
//mediapipe_api\mediapipe_proto_srcs.zip
DEBUG (build.py): Unarchiving '//mediapipe_api\mediapipe_proto_srcs.zip' to 'build\Scripts\Protobuf'...
Traceback (most recent call last):
  File "C:\Users\kmark\UnityProjects\MediaPipeUnityPlugin-master\build.py", line 460, in <module>
    Argument().command().run()
  File "C:\Users\kmark\UnityProjects\MediaPipeUnityPlugin-master\build.py", line 111, in run
    self._unzip(
  File "C:\Users\kmark\UnityProjects\MediaPipeUnityPlugin-master\build.py", line 96, in _unzip
    shutil.unpack_archive(source, dest)
  File "C:\Users\kmark\AppData\Local\Programs\Python\Python310\lib\shutil.py", line 1273, in unpack_archive
    func(filename, extract_dir, **kwargs)
  File "C:\Users\kmark\AppData\Local\Programs\Python\Python310\lib\shutil.py", line 1172, in _unpack_zipfile
    raise ReadError("%s is not a zip file" % filename)
shutil.ReadError: //mediapipe_api\mediapipe_proto_srcs.zip is not a zip file

**Comment by homuler** [comment]
> ran build command (path to python.exe build.py build --desktop cpu --opencv=cmake -vv

Is that all? Didn't you set an empty string to [`_BAZEL_BIN_PATH` in the `build.py`](https://github.com/homuler/MediaPipeUnityPlugin/blob/4302f4a174404ce5fff65fa2ce219ed71f8b9094/build.py#L16)?

**Comment by zigzagoon1** [comment]
> > ran build command (path to python.exe build.py build --desktop cpu --opencv=cmake -vv
> 
> Is that all? Didn't you set an empty string to [`_BAZEL_BIN_PATH` in the `build.py`](https://github.com/homuler/MediaPipeUnityPlugin/blob/4302f4a174404ce5fff65fa2ce219ed71f8b9094/build.py#L16)?

I had not. but just tried that and am still getting the same error.

**Comment by homuler** [comment]
>  just tried that

No, no, you MUST NOT do that. I thought that was the cause.

https://github.com/homuler/MediaPipeUnityPlugin/blob/4302f4a174404ce5fff65fa2ce219ed71f8b9094/build.py#L111

Your log says that `//mediapipe_api\mediapipe_proto_srcs.zip` ==  `os.path.join(_BAZEL_BIN_PATH, 'mediapipe_api', 'mediapipe_proto_srcs.zip')` but I think it can't be since `_BAZEL_BIN_PATH` is not empty (`bazel-bin`).
Do you know why it is in your environment?

**Comment by zigzagoon1** [comment]
So I realised I needed it to equal bazel-bin already, and I also figured out what the issue had been before and I'm embarrassed to admit it but it was totally my fault. I was having issues where it turned out cmd.exe was using python.exe from msys2 and it was therefore automatically changing // to / at one point and in my debugging efforts I added '//' before mediapipe_api in a few places and forgot to remove them again when it didn't work. Removing them fixed the error. So sorry to waste your time! 

I got the build to complete successfully, however, in Unity I am getting several errors about different types not having implicit reference conversions, ie. in BoxTrackingGraph.cs, _trackedDetectionsStream(Mediapipe.DetectionVectorPacket) cannot be implicitly converted to Mediapipe.packet<System.Collections.Generic.List<Mediapipe.Detection>>

**Comment by zigzagoon1** [comment]
They all seem to have to do with various classes not being able to be used as type parameter TPacket because of no reference conversion.

**Comment by zigzagoon1** [comment]
hmm. I extracted your repo to a new spot and created a new Unity project and redid the build command from that location, and when I opened the Unity project, I don't see any errors. I think I might have had Unity open when I built the libraries the first time, and Unity was in safe mode due to the compiler errors. So, again, embarrassing on my part, but I'm hoping it's maybe useful for you to know?
